LibraryApp/AccountActions.cs

```diff
@@ -0,0 +1,327 @@
+                        SetEmployeesData(person, employee);
+                    }
+                    // or register a reader
+                    else
+                    {
+                        SetReadersData(person);
+                    }
+
+                    SetAccountData(login, password, person);
+
+                    message = $"Аккаунт для {login} зарегистрирован";
+                }
+                else message = "Ошибка регистрации";
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show($"Не удалось создать новый аккаунт:\n\"{ex.Message}\"\n" +
+                $"Обратитесь к системному администратору для её устранения.",
+                "Ошибка при работе с базой данных", MessageBoxButtons.OK, MessageBoxIcon.Stop);
+            }
+
+            DataBase.CloseConnection();
+
+            return message;
+        }
+
+        // get account data
+        public Account GetCurrentAccountData(string login)
+        {
+            Account account = new();
+
+            string query = "SELECT * FROM Accounts WHERE Login=@login";
+
+            try
+            {
+                command = DataBase.GetConnection().CreateCommand();
+                command.CommandText = query;
+                command.Parameters.AddWithValue("@login", login);
+                DataBase.OpenConnection();
+                reader = command.ExecuteReader();
+                while (reader.Read())
+                {
+                    account = new Account
+                    {
+                        Id = reader.GetInt32(0),
+                        LoginId = reader.GetInt32(1),
+                        Login = reader.GetString(2),
+                        Password = reader.GetString(3),
+                        Salt = (byte[])reader[4]
+                    };
+                }
+                reader.Close();
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show($"Ошибка:\n\"{ex.Message}\"\n" +
+                                $"Обратитесь к системному администратору для её устранения.",
+                                "Ошибка при работе с базой данных", MessageBoxButtons.OK, MessageBoxIcon.Stop);
+            }
+
+            DataBase.CloseConnection();
+            return account;
+        }
+
+        // checking the presence of a login in the database before registration or authorization
+        public bool CheckInputedLogin(string login)
+        {
+            string query = "SELECT * FROM Accounts WHERE login = @login";
+
+            try
+            {
+                command = DataBase.GetConnection().CreateCommand();
+                command.CommandText = query;
+                command.Parameters.AddWithValue("@login", login);
+
+                DataBase.OpenConnection();
+                reader = command.ExecuteReader();
+
+                return reader.HasRows; // gets a value indicating whether the reader contains one or more lines
```

Comment:
When you return 'true' you're not closing the DB connection. Please, check the rest of the methods like this